app/controllers/application_controller.rb

```diff
@@ -111,4 +112,13 @@ def staff?
   def true_user
     @true_user ||= User.find_by(id: session[:user_id])
   end
+
+  def display_error_page(exception)
+    case exception
+    when ActionController::RoutingError
+      render file: Rails.root.join('public', '404.html'), layout: false
```

Comment:
This render line and the one below should have `, :status => :not_found` added. Otherwise, I believe rails will return a 200 response code.